pkg/segment/structs/querystructs.go

```diff
@@ -35,11 +35,12 @@ import (
 
 // New struct for passin query params
 type QueryContext struct {
-	TableInfo *TableInfo
-	SizeLimit uint64
-	Scroll    int
-	Orgid     uint64
-	RawQuery  string
+	TableInfo   *TableInfo
+	SizeLimit   uint64
+	Scroll      int
+	Orgid       uint64
+	RawQuery    string
+	SearchTerms map[string][]string
```

Comment:
nit: add a comment saying what this is mapping. Also are the search terms assumed to be joined by `AND` or `OR`, or does it not matter?